code/modules/modular_computers/hardware/job_disk.dm

```diff
@@ -229,3 +229,50 @@
 	name = "\improper FACEBUCKS disk"
 	icon_state = "cart-signal" // might need a new sprite
 	spam_delay = 5
+
+// maint spawn disks (including FACEBUCKS disk, and viruses)
+// Some of these are possibly inferior than its original version
+/obj/item/computer_hardware/hard_drive/role/random_maint_spawn
+	name = "random job disk"
+
+/obj/item/computer_hardware/hard_drive/role/random_maint_spawn/Initialize(mapload)
+	..()
+	var/static/disk_list = list(
+		/obj/item/computer_hardware/hard_drive/role/maint = 5, // value is pickweight
+		/obj/item/computer_hardware/hard_drive/role/medical = 1,
+		/obj/item/computer_hardware/hard_drive/role/chemistry = 1,
+		/obj/item/computer_hardware/hard_drive/role/signal/toxins = 1,
+		/obj/item/computer_hardware/hard_drive/role/security = 1,
+		/obj/item/computer_hardware/hard_drive/role/quartermaster = 1,
+		/obj/item/computer_hardware/hard_drive/role/virus/clown = 1,
+		/obj/item/computer_hardware/hard_drive/role/virus/mime = 1
+	)
+	var/static/adj_list = list("old", "weathered", "outdated")
+	var/obj/item/computer_hardware/hard_drive/role/chosen_disk = pickweight(disk_list)
+	chosen_disk = new chosen_disk(loc)
+	switch(chosen_disk.type) // let's not make much subtype items
+		if(/obj/item/computer_hardware/hard_drive/role/medical)
+			chosen_disk.name = "\improper [pick(adj_list)] Med-U disk"
```

Comment:
this is absolutely terrible, use the name already there

also if you want to remove different types of disk_flags just do it globally, remove ALL of them that you do not want to exist if possible

Reply:
These should absolutely be subtypes with all the special stuff done to them in init. This is a cluttered, hard to read mess.

Reply:
> These should absolutely be subtypes with all the special stuff done to them in init. 

they are difficult to be special since all of them are just identical but have fewer functions. I thought these are not good to be subtypes, so that's why these are chosen from a single subtype. I will follow a maintainer's opinion if they think it's needlessly cluttered too.